packages/x-internal-gestures/src/core/gestures/TapAndDragGesture.ts

```diff
@@ -0,0 +1,364 @@
+ *
+ * This gesture composes tap and drag logic patterns from TapGesture and PanGesture
+ * into a single coordinated gesture that handles tap-then-drag interactions.
+ */
+export class TapAndDragGesture<GestureName extends string> extends PointerGesture<GestureName> {
+  protected state: TapAndDragGestureState = {
+    phase: 'waitingForTap',
+    dragTimeoutId: null,
+  };
+
+  protected readonly isSinglePhase!: false;
+
+  protected readonly eventType!: TapAndDragEvent;
+
+  protected readonly optionsType!: TapAndDragGestureOptions<GestureName>;
+
+  protected readonly mutableOptionsType!: Omit<typeof this.optionsType, 'name'>;
+
+  protected readonly mutableStateType!: Omit<Partial<typeof this.state>, 'phase' | 'dragTimeoutId'>;
+
+  /**
+   * Maximum distance a pointer can move during tap for it to still be considered a tap
+   * (Following TapGesture pattern)
+   */
+  private tapMaxDistance: number;
+
+  /**
+   * Maximum time between tap completion and drag start
+   */
+  private dragTimeout: number;
+
+  /**
+   * Movement threshold for drag activation
+   * (Following PanGesture pattern)
+   */
+  private dragThreshold: number;
+
+  /**
+   * Allowed directions for the drag gesture
+   * (Following PanGesture pattern)
+   */
+  private dragDirection: Array<'up' | 'down' | 'left' | 'right'>;
+
+  /**
+   * Touch-action value to apply during the tap phase
+   */
+  private touchAction: string | undefined;
+
+  private tapGesture: TapGesture<GestureName>;
+
+  private panGesture: PanGesture<GestureName>;
+
+  private tapHandlerBound: () => void;
+
+  private dragStartHandlerBound: (event: PanEvent) => void;
+
+  private dragMoveHandlerBound: (event: PanEvent) => void;
+
+  private dragEndHandlerBound: (event: PanEvent) => void;
+
+  constructor(options: TapAndDragGestureOptions<GestureName>) {
+    super(options);
+    this.tapMaxDistance = options.tapMaxDistance ?? 10;
+    this.dragTimeout = options.dragTimeout ?? 1000;
+    this.dragThreshold = options.dragThreshold ?? 0;
+    this.dragDirection = options.dragDirection || ['up', 'down', 'left', 'right'];
+    this.tapGesture = new TapGesture({
+      name: `${this.name}-tap` as GestureName,
+      maxDistance: this.tapMaxDistance,
+      maxPointers: this.maxPointers,
+      pointerMode: this.pointerMode,
+      requiredKeys: this.requiredKeys,
+      preventIf: this.preventIf,
+      pointerOptions: structuredClone(this.pointerOptions),
+    });
+    this.panGesture = new PanGesture({
+      name: `${this.name}-pan` as GestureName,
+      minPointers: this.minPointers,
+      maxPointers: this.maxPointers,
+      threshold: this.dragThreshold,
+      direction: this.dragDirection,
+      pointerMode: this.pointerMode,
+      requiredKeys: this.requiredKeys,
+      preventIf: this.preventIf,
+      pointerOptions: structuredClone(this.pointerOptions),
+    });
+    this.tapHandlerBound = this.tapHandler.bind(this);
+    this.dragStartHandlerBound = this.dragStartHandler.bind(this);
```

Comment:
What is the benefit of explicit binding compared to arrow functions?
I don't think there is a big perf difference since @romgrk is using them in `useTimeout` for similar needs (https://github.com/mui/base-ui/blob/e51276c0ced4a3a0c106de53d147f2c8d9ecdf03/packages/utils/src/useTimeout.ts#L31)